cmd/generate/vault.go

```diff
@@ -35,17 +44,51 @@ func newVaultClient(config vaultClientConfig) (*vaultapi.Client, error) {
 	return vaultClient, nil
 }
 
+func createVaultClientUsingK8sSecret(ctx context.Context, namespace, name string) (*vaultapi.Client, error) {
+	c, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	client, err := kubernetes.NewForConfig(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	secret, err := client.CoreV1().Secrets(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	for _, varName := range []string{vaultAddress, vaultToken, vaultCAPath} {
+		if value, ok := secret.Data[varName]; !ok || string(value) == "" {
+			return nil, microerror.Maskf(executionFailedError, "secret.Data must contain %q", varName)
+		}
+	}
+
+	vaultClient, err := newVaultClient(vaultClientConfig{
+		Address: string(secret.Data[vaultAddress]),
+		Token:   string(secret.Data[vaultToken]),
+		CAPath:  string(secret.Data[vaultCAPath]),
```

Comment:
I think it would be good to check if those are not empty and return a nicer error if so.